akd/src/directory.rs

```diff
@@ -334,6 +331,74 @@ impl<S: Storage + Sync + Send, V: VRFKeyStorage> Directory<S, V> {
         }
     }
 
+    /// Takes in the current state of the server and a label along with
+    /// a range starting epoch.
+    ///
+    /// If the label is present in the current state,
+    /// this function returns all the values ever associated with it after start_epoch,
+    /// and the epoch at which each value was first committed to the server state.
+    /// It also returns the proof of the latest version being served at all times.
+    pub async fn limited_key_history<H: Hasher>(
+        &self,
+        start_epoch: u64,
+        uname: &AkdLabel,
+    ) -> Result<HistoryProof<H>, AkdError> {
+        // The guard will be dropped at the end of the proof generation
+        let _guard = self.cache_lock.read().await;
+
+        let username = uname.to_vec();
+        let current_azks = self.retrieve_current_azks().await?;
+        let current_epoch = current_azks.get_latest_epoch();
+
+        if start_epoch >= current_epoch {
+            return Err(AkdError::Directory(DirectoryError::InvalidEpoch(format!(
+                "start_epoch ({}) needs to be before the current epoch {}",
+                start_epoch, current_epoch
+            ))));
+        }
+
+        // TODO: This call is quite expensive, since it's assumed most epochs do NOT have a
```

Comment:
This issue was already solved in the original way key history was done. Essentially you want to get a user's version numbers at `start_epoch` and `end_epoch` and prove history proofs like below. Is there a reason you traverse over epochs as if they are version numbers?

Reply:
So I understand what you're getting at, but it still pulled the _entire_ history for an entry. Imagine the following scenario:

1. User has account, does 10 key updates
2. User deletes account, we tombstone 10 updates
3. New user gets phone number of User in (1), they do 10 key updates

Now a key-history for the "new" user in (3) will show 20 entries. Now scale this to 100 updates, over 10 users, etc. Eventually we might not want _all_ of the key history for a user, but say the history in the last X epochs. That's what this call is solving. So it's a "truncated" history, for a specific range of values. Does that make sense?